### PR TITLE
Add support for the `@Image` directive

### PR DESCRIPTION
**Pull request by dobromir-hristov**
Bug/issue #, if applicable: 97715120

## Summary

Extends support for the `@Image` directive, by allowing users to add captions to images. 

image

It uses the already available `InlineImage` component, and just extends the `Figure` logic to wrap, if either an `anchor` or `abstract` is present.

## Dependencies

DocC PR - 

## Testing

You can add this RenderJSON piece to the `content` array of your render json `primaryContentSections`:

```json
                {
                  "type": "image",
                  "identifier": "reference-for-image",
                  "metadata": {
                    "abstract": [
                      {
                        "type": "text",
                        "text": "Image Caption"
                      }
                    ]
                  }
                }
```

Or use the attached JSON:

[example_image.json.zip](https://github.com/apple/swift-docc-render/files/9289833/example_image.json.zip)

Steps:
1. Assert the caption is rendered below the image

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

File: src/components/ContentNode/FigureCaption.vue [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 <template>
   <figcaption class="caption">
-    <strong>{{ title }}</strong>&nbsp;<slot />
+    <strong v-if="title">{{ title }}</strong>&nbsp;<slot />
```
Comment (mportiz08) [inline src/components/ContentNode/FigureCaption.vue:13]: Should this component and its `<figcaption>` wrapper ever even render if it has no text content to show?

File: src/components/ContentNode.vue [hunk 2]
```diff
@@ -284,7 +285,7 @@ function renderNode(createElement, references) {
         renderChildren(node.inlineContent)
       ));
     case InlineType.image: {
-      if (node.metadata && node.metadata.anchor) {
+      if (node.metadata && (node.metadata.anchor || node.metadata.abstract)) {
```
Comment (mportiz08) [inline src/components/ContentNode.vue:288]: Has the DocC RenderNode spec been updated to reflect this? (Just checking)

File: src/components/ContentNode/FigureCaption.vue (hunk 1 above)
Comment (dobromir-hristov) [reply src/components/ContentNode/FigureCaption.vue:13]: My idea was to still render a `<figcaption>` and populate it via the `<slot>`, I just `v-if` the bolded `title`, which this feature will not utilise.

File: src/components/ContentNode.vue (hunk 2 above)
Comment (dobromir-hristov) [reply src/components/ContentNode.vue:288]: Ethan showed me a draft spec. I hope we get some actual DocC PRs in the next few days.

File: src/components/ContentNode/FigureCaption.vue (hunk 1 above)
Comment (mportiz08) [reply src/components/ContentNode/FigureCaption.vue:13]: Oh, got it. I thought the title was all the text and missed the slot—I'm remembering that those are 2 separate things (title and content).

**Review by mportiz08 — APPROVED** [review]
Tested with the DocC branch and looks good (no actual spec changes needed for the image specifically it seems).

The only potential issue I noticed is that the proposal for this new directive showed examples where the caption text is centered underneath the image as opposed to left aligned as it is now. If we were to support that, we may need to update the directive itself to allow for both alignments (because we already support the left aligned ones for older content).

**Comment by dobromir-hristov** [comment]
@swift-ci test

**Pull request merged**